src/decode.rs

```diff
@@ -163,3 +265,34 @@ mod tests {
         assert_eq!(decode("a3gV").into(&mut output), Err(DecodeError::BufferTooSmall));
     }
 }
+
+#[cfg(test)]
+#[cfg(feature = "check")]
+mod test_check{
+    use decode;
+    use decode::DecodeError;
+
+    #[test]
+    fn test_check(){
+        for &(val, s) in super::super::CHECK_TEST_CASES.iter() {
+            assert_eq!(val.to_vec(), decode(s).with_check(None).into_vec().unwrap());
+        }
+
+        for &(val, s) in super::super::CHECK_TEST_CASES[1..].iter() {
+            assert_eq!(val.to_vec(), decode(s).with_check(Some(val[0])).into_vec().unwrap());
+        }
+    }
+
+    #[test]
+    fn test_check_ver_failed() {
+        let d = decode(super::super::CHECK_TEST_CASES[6].1)
```

Comment:
I'd prefer to just copy the test string into this function, to avoid this being accidentally broken if the test cases are changed in the future, and being a little more explicit about what is tested here.

Reply:
will change

Reply:
Done